examples/hosts/app-integration/schema-upgrade/src/migrator/migrator.ts

```diff
@@ -131,39 +139,12 @@ export class Migrator extends TypedEventEmitter<IMigratorEvents> implements IMig
             }
             await migratedModel.importData(transformedData);
 
-            // Before attaching, let's check to make sure no one else has already done the migration
-            // To avoid creating unnecessary extra containers.
-            if (migratable.migrationTool.migrationState === "migrated") {
-                this._migrationP = undefined;
-                migratedModel.close();
-                this.takeAppropriateActionForCurrentMigratable();
-                return;
-            }
-
             // TODO: Support retry
-            // TODO: Use TaskManager here to reduce container noise.  Specifically -- all clients should race up to
-            // this point (so they're all prepared to do the migration) but should wait for the task lock before
-            // attempting the attach() to minimize the chance that multiple containers are created on the service.
             const containerId = await createResponse.attach();
```

Comment:
It's OK for all clients to simultaneously do everything up until the attach call.  However, since it's currently all in doTheMigration, clients will only do the export/transform/import-into-detached once they get the task, which will be slower if a second client needs to take over.  Consider splitting this into two chunks - preparing for the migration (not guarded by TaskManager) vs. attach/finalize (guarded by task manager)